File: app/src/main/java/kr/kro/awesometic/plankhelper/statistics/log/LogContract.java
package kr.kro.awesometic.plankhelper.statistics.log;

import kr.kro.awesometic.plankhelper.BasePresenter;
import kr.kro.awesometic.plankhelper.BaseView;

/**
 * Created by Awesometic on 2017-05-17.
 */

public interface LogContract {

    interface View extends BaseView<Presenter> {

        void showLoading();
        void showLog();

        Object getActivityContext();
        void setRecyclerViewAdapter(Object recyclerViewAdapter);

        void bindViewsFromViewHolder();
    }

    interface Presenter extends BasePresenter {
        void bindViewsFromViewHolderToFrag();
    }
}

File: app/src/main/java/kr/kro/awesometic/plankhelper/statistics/log/LogContract.kt
package kr.kro.awesometic.plankhelper.statistics.log

import kr.kro.awesometic.plankhelper.BasePresenter
import kr.kro.awesometic.plankhelper.BaseView

/**
 * Created by Awesometic on 2017-05-17.
 */

interface LogContract {

    interface View : BaseView<Presenter> {

        val applicationContext: Any

        fun showLoading()
        fun showLog()
        fun setRecyclerViewAdapter(recyclerViewAdapter: Any)

        fun bindViewsFromViewHolder()
    }

    interface Presenter : BasePresenter {
        fun bindViewsFromViewHolderToFrag()
    }
}
